### ISSUE TITLE
[Feature-Request]: Add `Version` number to be fetched from each smart contracts

### ISSUE THREAD

**Issue by CJ42** [lukso-network/lsp-smart-contracts#756]
### Describe the desired feature:

Currently, on each new releases, big parts of the inner code logic changes. For instance:

- new features get added (e.g: notifying LSP1 on `renounceOwnership`, adding path to the Key Manager, etc...)
- bug are fixed 
- breaking changes in the inner logic are introduced (e.g: `universalReceiver(...)` function in LSP0Core now checks for `LSP1_DELEGATE` interface ID).

For UIs and interfaces, there is no way to know if the contract supports some features, implement a fix, or has a different inner logic that will make calling the contract work or not. 

This is necessary for UIs to know what to display. For instance:
- if the LSP0ERC725Account of the user can be linked to a particular `LSP1Delegate` contract. If the UP was deployed with v0.11.0, the LSP1 Delegate would have to implements `universalReceiver(bytes32,bytes)`, while if the UP was deployed with v0.12.0, it would need to be a contract that implements `universalReceiverDelegate(...)`.

- if the LSP0 of the user can be upgraded to a new and better Key Manager. A UP with version v0.12 **would need to have this upgraded Key Manager to implement the new LSP20 interface from v0.12 with all its new functions parameters `lsp20VerifyCall(address,address,address,uint256,bytes)**. Otherwise, any LSP20 verification logic will fail.

## Rationale

Having the version number stored in the smart contract is important, so that applications that deployed LSP smart contracts with former versions can know if they can link / upgrade their:
- LSP6 Key Manager
- LSP1 Delegate
- LSP17 Extensions

To new contracts deployed with these new versions.

## Note on interface IDs

The interface ID cannot be used reliably to check for these things.

From v0.11 to v0.12, the interface ID of LSP0 did not changed, but actually the inner logic of the contract completely changed, which makes it incompatible (check for LSP1 Delegate interface ID, not LSP1 interface ID to call the Universal Receiver Delegate + call to the owner via LSP20 completely changed, as we now have more parameters added to the function lsp20VerifyCall and the LSP0 contract makes this change).

The other way around, from v0.10 to v0.11, the interface ID of LSP0 changed, but actually the functions from the interface did not change at all (we just added LSP20 in the calculation).
So you think it is not compatible anymore, while it actually is completely compatible.

### Code example that solves the feature:

```solidity
abstract contract Version {
    string public constant VERSION = "0.12.0";
}

contract LSP0ERC725AccountCore is Version, ... {
    // all the remaining Solidity code
}
```

**Comment by skimaharvey** [comment]
I agree. I think @YamenMerhi wanted to propose this idea at some point. Just not sure if we could not use the `ERC725Y`storage for that (would not work for some contract though)

**Comment by CJ42** [comment]
Closing as implemented in #791
